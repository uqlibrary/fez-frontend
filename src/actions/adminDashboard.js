import * as actions from './actionTypes';
import { get, post, put, destroy } from 'repositories/generic';
import {
    ADMIN_DASHBOARD_CONFIG_API,
    ADMIN_DASHBOARD_TODAY_API,
    ADMIN_DASHBOARD_QUICKLINKS_API,
    ADMIN_DASHBOARD_SYSTEM_ALERTS_API,
    ADMIN_DASHBOARD_DISPLAY_REPORT_API,
    ADMIN_DASHBOARD_EXPORT_REPORT_API,
} from 'repositories/routes';

import { promptForDownload } from './exportPublicationsDataTransformers';

/**
 * Fetches the config data for admin dashboard
 * @returns {function(*)}
 */
export function loadAdminDashboardConfig() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_CONFIG_LOADING,
        });
        return get(ADMIN_DASHBOARD_CONFIG_API())
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_CONFIG_SUCCESS,
                    payload: response,
                });
                return Promise.resolve(response);
            })
            .catch(error => {
                console.error(error.message);
                dispatch({
                    type: actions.ADMIN_DASHBOARD_CONFIG_FAILED,
                    payload: error.message,
                });
                return Promise.reject(error);
            });
    };
}

/**
 * Fetches the metrics for the Today tab
 * @returns {function(*)}
 */
export function loadAdminDashboardToday() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_TODAY_LOADING,
        });
        return get(ADMIN_DASHBOARD_TODAY_API())
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_TODAY_SUCCESS,
                    payload: response,
                });
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_TODAY_FAILED,
                    payload: error.message,
                });
            });
    };
}

/**
 * Fetches the quicklinks
 * @returns {function(*)}
 */
export function loadAdminDashboardQuickLinks() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_QUICKLINKS_LOADING,
        });
        return get(ADMIN_DASHBOARD_QUICKLINKS_API())
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_QUICKLINKS_SUCCESS,
                    payload: response,
                });
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_QUICKLINKS_FAILED,
                    payload: error.message,
                });
            });
    };
}

/**
 * Handles admin operations on quick links (edit, delete)
 * @returns {function(*)}
 */
export function adminDashboardQuickLink(request, action) {
    // eslint-disable-next-line no-nested-ternary
    const verb = action === 'DELETE' ? destroy : action === 'EDIT' || action === 'REORDER' ? put : post;

    return dispatch => {
        dispatch({ type: actions.ADMIN_DASHBOARD_QUICKLINKS_UPDATING });

        return verb(ADMIN_DASHBOARD_QUICKLINKS_API(), request)
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_QUICKLINKS_UPDATE_SUCCESS,
                    payload: response,
                });
                return Promise.resolve(response);
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_QUICKLINKS_UPDATE_FAILED,
                    payload: error.message,
                });
                return Promise.reject(error);
            });
    };
}

export function adminDashboardQuickLinkUpdateClear() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_QUICKLINKS_UPDATE_CLEAR,
        });
    };
}

/**
 * Fetches the System Alerts
 * @returns {function(*)}
 */
export function loadAdminDashboardSystemAlerts() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_SYSTEM_ALERTS_LOADING,
        });
        return get(ADMIN_DASHBOARD_SYSTEM_ALERTS_API())
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_SYSTEM_ALERTS_SUCCESS,
                    payload: response,
                });
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_SYSTEM_ALERTS_FAILED,
                    payload: error.message,
                });
            });
    };
}

/**
 * Handles admin operations on system alerts (assign, resolve)
 * @returns {function(*)}
 */
export function adminDashboardSystemAlerts(request) {
    return dispatch => {
        dispatch({ type: actions.ADMIN_DASHBOARD_SYSTEM_ALERT_UPDATING });
        return put(ADMIN_DASHBOARD_SYSTEM_ALERTS_API(), request)
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_SYSTEM_ALERT_UPDATE_SUCCESS,
                    payload: response,
                });

                return Promise.resolve(response);
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_SYSTEM_ALERT_UPDATE_FAILED,
                    payload: error.message,
                });
                return Promise.reject(error);
            });
    };
}

export function adminDashboardSystemAlertsUpdateClear() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_SYSTEM_ALERTS_UPDATE_CLEAR,
        });
    };
}

/**
 * Fetches export (legacy) report data as a file attachment
 * or queues the request at the server.
 * @returns {function(*)}
 */
export function loadAdminDashboardExportReport(request, options) {
    return dispatch => {
        const exportConfig = {
            format: options.export_to,
        };

        dispatch({
            type: actions.ADMIN_DASHBOARD_EXPORT_REPORT_LOADING,
        });

        return get(ADMIN_DASHBOARD_EXPORT_REPORT_API(request))
            .then(response => {
                // only try to export to csv if we get a text response
                if (typeof response === 'string') {
                    const blobResponse = new Blob([response], { type: 'text/csv' });
                    promptForDownload(exportConfig.format, blobResponse);
                }

                dispatch({
                    type: actions.ADMIN_DASHBOARD_EXPORT_REPORT_SUCCESS,
                    payload: exportConfig,
                });

                return Promise.resolve(response);
            })
            .catch(error => {
                console.error(error);
                dispatch({
                    type: actions.ADMIN_DASHBOARD_EXPORT_REPORT_FAILED,
                    payload: {
                        ...exportConfig,
                        errorMessage: error.message,
                    },
                });
            });
    };
}

/**
 * Fetches display report data
 * @returns {function(*)}
 */
export function loadAdminDashboardDisplayReport(request) {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_DISPLAY_REPORT_LOADING,
            params: { ...request },
        });
        return get(ADMIN_DASHBOARD_DISPLAY_REPORT_API(request))
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_DISPLAY_REPORT_SUCCESS,
                    payload: response,
                });
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_DISPLAY_REPORT_FAILED,
                    payload: {
                        errorMessage: error.message,
                    },
                });
            });
    };
}

export function clearAdminDashboardDisplayReport() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_DISPLAY_REPORT_CLEAR,
        });
    };
}

export function clearAdminDashboardExportReport() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_EXPORT_REPORT_CLEAR,
        });
    };
}
