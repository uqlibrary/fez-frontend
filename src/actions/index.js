export * from './academic';
export * from './account';
export * from './app';
export * from './authors';
export * from './batchImport';
export * from './bulkUpdates';
export * from './claimPublications';
export * from './controlledVocabularies';
export * from './deleteRecord';
export * from './doi';
export * from './exportPublications';
export * from './favouriteSearch';
export * from './fixRecord';
export * from './incompleteRecords';
export * from './issnLinks';
export * from './journals';
export * from './manageAuthors';
export * from './manageUsers';
export * from './myEditorialAppointments';
export * from './orcidSync';
export * from './organisationalUnits';
export * from './publications';
export * from './records';
export * from './search';
export * from './thirdPartyLookupTool';
export * from './viewRecord';
export * from './viewCommunitiesCollections';
export * from './adminScaleSignificance';
export * from './viewControlledVocab';
