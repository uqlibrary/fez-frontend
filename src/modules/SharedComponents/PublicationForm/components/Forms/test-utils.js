import { useForm } from 'react-hook-form';
import { render, WithReduxStore } from 'test-utils';
import React from 'react';
import { formValues } from '../PublicationForm';

export const setupWrapper = (Form, props) => {
    const Wrapper = () => {
        const {
            control,
            formState: { isSubmitting },
        } = useForm();

        // eslint-disable-next-line react/prop-types
        return <Form {...{ control, isSubmitting, ...props, formValues: formValues(props?.formValues || {}) }} />;
    };

    return render(
        <WithReduxStore>
            <Wrapper />
        </WithReduxStore>,
    );
};
