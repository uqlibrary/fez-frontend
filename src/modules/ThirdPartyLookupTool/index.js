export { default as ThirdPartyLookupTool } from './components/ThirdPartyLookupTool';
