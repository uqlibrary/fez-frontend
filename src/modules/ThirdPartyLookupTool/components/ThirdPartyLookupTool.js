import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { locale } from 'locale';

import { useSelector } from 'react-redux';
import * as actions from 'actions';

import { StandardPage } from 'modules/SharedComponents/Toolbox/StandardPage';
import { InlineLoader } from 'modules/SharedComponents/Toolbox/Loaders';

import { ThirdPartyLookupForm } from './ThirdPartyLookupForm';
import { ThirdPartyLookupFormResult } from './ThirdPartyLookupFormResult';

/**
 * This component is intended to replace https://espace.library.uq.edu.au/misc/api_view_info.php
 * but only an additional form (incites) has been done at this point
 * As each form comes in:
 * - add an entry to locale/components.js under .thirdPartyLookupTools.forms as a sibling to .incites;
 *          - match the contents of the incites entry
 *          - the secondary field is optional
 *          - add a new form entry below (once we have more than one component
 *          use the default isMinimised value so they load minimised)
 * - create a new api (use FezCoreLookupController) that returns the desired data, with a url of:
 *    tool/lookup/[components.thirdPartyLookupTools.forms.sibling]/[keys to search for]/[possible other field you need]
 *    ie the api url that incites calls is: eg
 *    "https://api.library.uq.edu.au/staging/tool/lookup/incites/A1979HY31900010,A1979HY31900068/[APIKEY]".
 * - and that should be all that is required (but do examine how the result is displayed...)
 */
export const ThirdPartyLookupTool = () => {
    const [state, setState] = React.useState({
        primaryValue: '',
        secondaryValue: '',
        formDisplay: {},
    });

    const { loadingResults, lookupResults } = useSelector(state => state.get('thirdPartyLookupToolReducer'));

    const recordInputs = (primaryValue, secondaryValue, formDisplay) => {
        setState({ primaryValue, secondaryValue, formDisplay });
    };

    const localeContent = locale.components.thirdPartyLookupTools.display;
    return (
        <StandardPage title={localeContent.title}>
            {loadingResults && <InlineLoader message={localeContent.loadingMessage} />}
            {// this still needs work because we have to pass the specific form details for display
            // pass through with the results? or via sendInputsToResultComponent ?
            !loadingResults && !!lookupResults && lookupResults.length > 0 && (
                <ThirdPartyLookupFormResult
                    lookupResults={lookupResults}
                    actions={actions}
                    formDisplay={state.formDisplay}
                    primaryValue={state.primaryValue}
                    secondaryValue={state.secondaryValue}
                    locale={localeContent}
                />
            )}

            {!loadingResults &&
                !!lookupResults &&
                lookupResults.length === 0 &&
                !!locale.components.thirdPartyLookupTools.forms &&
                locale.components.thirdPartyLookupTools.forms.length > 0 && (
                    <Fragment>
                        {locale.components.thirdPartyLookupTools.forms.map(form => (
                            <ThirdPartyLookupForm
                                key={form.apiType}
                                locale={localeContent}
                                localeform={form}
                                actions={actions}
                                sendInputsToResultComponent={recordInputs} // function
                                isMinimised={!!form.isMinimised}
                            />
                        ))}
                    </Fragment>
                )}
        </StandardPage>
    );
};
ThirdPartyLookupTool.propTypes = {
    actions: PropTypes.object,
    lookupResults: PropTypes.array,
    loadingResults: PropTypes.bool,
};
ThirdPartyLookupTool.defaultProps = {
    lookupResults: [],
    loadingResults: false,
};

export default React.memo(ThirdPartyLookupTool);
