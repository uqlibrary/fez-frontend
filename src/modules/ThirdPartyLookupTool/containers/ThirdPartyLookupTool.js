import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actions from 'actions';

import ThirdPartyLookupTool from '../components/ThirdPartyLookupTool';

const mapStateToProps = state => {
    return {
        ...state.get('thirdPartyLookupToolReducer'),
    };
};

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch),
    };
}

const ThirdPartyLookupContainer = connect(mapStateToProps, mapDispatchToProps)(ThirdPartyLookupTool);

export default ThirdPartyLookupContainer;
