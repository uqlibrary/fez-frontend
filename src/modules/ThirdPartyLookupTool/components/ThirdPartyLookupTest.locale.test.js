import React from 'react';
import Immutable from 'immutable';

import { rtlRender, fireEvent, WithReduxStore } from 'test-utils';
import { ThirdPartyLookupTool } from './ThirdPartyLookupTool';
import * as actions from 'actions';

jest.mock('actions', () => ({
    ...jest.requireActual('actions'),
    loadThirdPartyResults: jest.fn(() => jest.fn()),
}));

jest.mock('locale', () => ({
    locale: {
        components: {
            thirdPartyLookupTools: {
                display: {
                    title: 'Lookup Tools - view raw output from APIs',
                    loadingMessage: 'Loading',
                    tooltip: {
                        show: 'Show form for',
                        hide: 'Hide form for',
                    },
                    resultsLabel: 'Results',
                    noResultsFound: {
                        text: 'No results found',
                    },
                    clearButtonLabel: 'New Search',
                },
                forms: [
                    {
                        apiType: 'incites', // this value should match the 'type' in the path used in api
                        lookupLabel: 'Incites',
                        primaryField: {
                            heading: 'UTs',
                            fromAria: 'primaryValue',
                            tip: '',
                            inputPlaceholder: 'Enter one or more UTs separated by a comma e.g. 000455548800001',
                        },
                        secondaryField: {
                            heading: 'API Key',
                            fromAria: 'secondaryValue',
                            tip: 'Optional, a default key is provided. Limit: 1,000 queries per day',
                            inputPlaceholder: 'Enter API key',
                            reportInOutput: false, // determines if secondaryField will apear in the results page
                        },
                        bottomTip: '',
                        submitButtonLabel: 'Submit to Incites',
                        isMinimised: false, // set this to false when we have more than one form
                    },
                ],
            },
        },
    },
}));

function setup(testProps = {}, testState = {}) {
    const props = {
        ...testProps,
    };
    const state = {
        thirdPartyLookupToolReducer: {
            ...testState,
        },
    };

    return rtlRender(
        <WithReduxStore initialState={Immutable.Map(state)}>
            <ThirdPartyLookupTool {...props} />
        </WithReduxStore>,
    );
}

describe('Component ThirdPartyLookupTool', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render a form ready for input', () => {
        const { container } = setup({}, { lookupResults: [] });
        expect(container).toMatchSnapshot();
    });

    it('should set state with submitted data', () => {
        const { container, getByRole } = setup({}, { lookupResults: [] });

        expect(getByRole('textbox', { name: 'primaryValue' }).value).toEqual('');
        expect(getByRole('textbox', { name: 'secondaryValue' }).value).toEqual('');

        expect(container).toMatchSnapshot();

        fireEvent.change(getByRole('textbox', { name: 'primaryValue' }), { target: { value: 'a value' } });
        fireEvent.change(getByRole('textbox', { name: 'secondaryValue' }), { target: { value: 'another value' } });

        expect(getByRole('button', { name: 'Submit to Incites' })).toBeInTheDocument();
        fireEvent.click(getByRole('button', { name: 'Submit to Incites' }));

        expect(getByRole('textbox', { name: 'primaryValue' }).value).toEqual('a value');
        expect(getByRole('textbox', { name: 'secondaryValue' }).value).toEqual('another value');

        expect(actions.loadThirdPartyResults).toHaveBeenCalled();
    });

    it('renders loading screen while loading data', () => {
        const { container } = setup({}, { loadingResults: true });
        expect(container).toMatchSnapshot();
    });

    it('renders a results screen', () => {
        const { container } = setup({}, { lookupResults: ['blah blah blah'] });
        expect(container).toMatchSnapshot();
    });
});
