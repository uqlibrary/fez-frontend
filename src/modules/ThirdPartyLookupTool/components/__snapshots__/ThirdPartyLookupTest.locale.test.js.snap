// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ThirdPartyLookupTool renders a results screen 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container StandardPage css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-body1 css-v2c8i0-MuiTypography-root"
        data-analyticsid="page-title"
        data-testid="page-title"
        id="page-title"
      >
        Lookup Tools - view raw output from APIs
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1guqyqn-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root StandardCard css-985319-MuiPaper-root-MuiCard-root"
            data-testid="standard-card"
            id="standard-card"
          >
            <div
              class="MuiCardHeader-root css-16c6n48-MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
              />
            </div>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              data-analyticsid="standard-card-content"
              data-testid="standard-card-content"
              id="standard-card-content"
            >
              <p>
                 - 
              </p>
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root StandardCard css-985319-MuiPaper-root-MuiCard-root"
                data-testid="standard-card-results"
                id="standard-card-results"
              >
                <div
                  class="MuiCardHeader-root css-16c6n48-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <h3
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1hzv55z-MuiTypography-root"
                      data-testid="standard-card-results-header"
                      id="standard-card-results-header"
                    >
                      Results
                    </h3>
                  </div>
                </div>
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                  data-analyticsid="standard-card-results-content"
                  data-testid="standard-card-results-content"
                  id="standard-card-results-content"
                >
                  <pre>
                    [
  "blah blah blah"
]
                  </pre>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-argx73-MuiButtonBase-root-MuiButton-root"
                style="margin-top: 20px;"
                tabindex="0"
                type="button"
              >
                New Search
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component ThirdPartyLookupTool renders loading screen while loading data 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container StandardPage css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-body1 css-v2c8i0-MuiTypography-root"
        data-analyticsid="page-title"
        data-testid="page-title"
        id="page-title"
      >
        Lookup Tools - view raw output from APIs
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1guqyqn-MuiGrid-root"
    >
      <div
        style="padding: 8px;"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-qiawb8-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-5os2wf-MuiGrid-root"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
            style="text-align: center;"
          >
            <span
              aria-label="Loading"
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-u4ps4u-MuiCircularProgress-root"
              role="progressbar"
              style="width: 18px; height: 18px;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="21"
                  stroke-width="2"
                />
              </svg>
            </span>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
            style="text-align: center;"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 css-1q4fwm5-MuiTypography-root"
              style="font-size: 1.33rem;"
            >
              Loading
            </span>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-5os2wf-MuiGrid-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component ThirdPartyLookupTool should render a form ready for input 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container StandardPage css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-body1 css-v2c8i0-MuiTypography-root"
        data-analyticsid="page-title"
        data-testid="page-title"
        id="page-title"
      >
        Lookup Tools - view raw output from APIs
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1guqyqn-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root StandardCard css-985319-MuiPaper-root-MuiCard-root"
            data-testid="standard-card"
            id="standard-card"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              data-analyticsid="standard-card-content"
              data-testid="standard-card-content"
              id="standard-card-content"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  style="flex-grow: 1; width: 1px;"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-19cmg10-MuiTypography-root"
                  >
                    Incites
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-bc0omg-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="minimise-toggle"
                    tabindex="0"
                    tooltip="Hide form for Incites"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowUpIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <form>
                <p />
                <div>
                  <h4>
                    UTs
                  </h4>
                  <p />
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root primaryValue css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1vg8b0p-MuiInputBase-root-MuiInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="primaryValue"
                        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                        id=":r0:"
                        name="primaryValue"
                        placeholder="Enter one or more UTs separated by a comma e.g. 000455548800001"
                        required=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div>
                  <h4>
                    API Key
                  </h4>
                  <p>
                    Optional, a default key is provided. Limit: 1,000 queries per day
                  </p>
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root secondaryValue css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1vg8b0p-MuiInputBase-root-MuiInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="secondaryValue"
                        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                        id=":r1:"
                        name="secondaryValue"
                        placeholder="Enter API key"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <p />
                <button
                  aria-label="Submit to Incites"
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-argx73-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Submit to Incites
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component ThirdPartyLookupTool should set state with submitted data 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container StandardPage css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-body1 css-v2c8i0-MuiTypography-root"
        data-analyticsid="page-title"
        data-testid="page-title"
        id="page-title"
      >
        Lookup Tools - view raw output from APIs
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1guqyqn-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root StandardCard css-985319-MuiPaper-root-MuiCard-root"
            data-testid="standard-card"
            id="standard-card"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              data-analyticsid="standard-card-content"
              data-testid="standard-card-content"
              id="standard-card-content"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  style="flex-grow: 1; width: 1px;"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-19cmg10-MuiTypography-root"
                  >
                    Incites
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-bc0omg-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="minimise-toggle"
                    tabindex="0"
                    tooltip="Hide form for Incites"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowUpIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <form>
                <p />
                <div>
                  <h4>
                    UTs
                  </h4>
                  <p />
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root primaryValue css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1vg8b0p-MuiInputBase-root-MuiInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="primaryValue"
                        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                        id=":r2:"
                        name="primaryValue"
                        placeholder="Enter one or more UTs separated by a comma e.g. 000455548800001"
                        required=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div>
                  <h4>
                    API Key
                  </h4>
                  <p>
                    Optional, a default key is provided. Limit: 1,000 queries per day
                  </p>
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root secondaryValue css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1vg8b0p-MuiInputBase-root-MuiInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="secondaryValue"
                        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                        id=":r3:"
                        name="secondaryValue"
                        placeholder="Enter API key"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <p />
                <button
                  aria-label="Submit to Incites"
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-argx73-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Submit to Incites
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
